orchagent/portsorch.cpp

```diff
@@ -1045,7 +1051,21 @@ void PortsOrch::initializeQueues(Port &port)
         SWSS_LOG_ERROR("Failed to get queue list for port %s rv:%d", port.m_alias.c_str(), status);
         throw runtime_error("PortsOrch initialization failure.");
     }
+
     SWSS_LOG_INFO("Get queues for port %s", port.m_alias.c_str());
+
+    /* Create the Queue map in the Counter DB */
+    vector<FieldValueTuple> vector;
+
+    for (size_t queue_index = 0; queue_index < port.m_queue_ids.size(); ++queue_index)
+    {
+        std::stringstream name;
+        name << port.m_alias << ":" << queue_index;
+        FieldValueTuple tuple(name.str(), sai_serialize_object_id(port.m_queue_ids[queue_index]));
+        vector.push_back(tuple);
+    }
+
+    m_queueTable->set("", vector);
```

Comment:
design question: why serialize all pairs into 'vector'? why not store them as key-value in the table?

Reply:
It's written in the same style as port counters